tidy-clustering/README.Rmd

```diff
@@ -0,0 +1,222 @@
+
+I don't really see a parallel between the classification/regression divide and the partition/extraction divide.  It doesn't feel right to just lump them together into four "modes" options.
+
+#### Choosing a dissimilarity metric
+
+Every model comes with choices about parameters and assumptions and such.  To some degree, this would work the same in `parsnip` and `celery`:
+
+```{r, eval = FALSE}
+knn_mod <- nearest_neighbors(neighbors = 5)
+
+km_mod <- k_means(centers = 5)
+```
+
+However, for clustering, the decision of how to define "dissimilarity" feels like a more major decision.  That is, supervised methods require **one major choice** (what model) and then many minor choices (the parameters/penalties) for that model. Unsupervised methods require **two major choices**: the algorithm and the similarity measure.
+
+I would like to see the dissimilarity measure be given "front-row billing", to
+
+a. make it easy to swap between choices and compare outputs
+b. make it obvious what we are looking for in our clusters
+c. force the user to choose very deliberately rather than relying on defaults
+
+For example,
+
+```{r, eval = FALSE}
+k_means(centers = 5) %>%
+    set_dist("euclidean")
+
+k_means(centers = 5) %>%
+    set_dist("manhattan")
+```
+
+Much like with *modes*, this would require the `celery` package to pre-specify which disssimilarities were available for particular clustering algorithms.  For example, some methods require a true distance measure, while others are less strict.
+
+These would have to be more flexible than modes, though, since clustering methods have **so** many possible dissimilarity measures.
+
+## fitting to data
+
+There is no real reason the `fit` step needs to look different, once a model is specified:
+
+```{r, eval = FALSE}
+
+km_mod <- k_means(centers = 5) %>%
+    set_dist("euclidean") %>%
+    set_mode("partition") %>%
+    set_engine("stats")
+
+
+km_mod %>%
+    fit(vars(bill_length_mm:body_mass_g), data = penguins)
+```
+
+It's been suggested that "fit" might not be the right verb for unsupervised methods.  Technicallly, many supervised methods are not truly fitting a model either (see: K nearest neighbors), so I'm not worried about this.
+
+## `rsample` and cross-validating
+
+Cross-validation as such doesn't make sense for clustering, because there is no notion of prediction on the test set.
+
+However, a similar framework could make sense for subsampling.  Instead of $v$ cross-validations, we'd have $v$ subsamples, and we'd look at certain success metrics for each subsample to get a sense of variability in the metric.
+
+
+## `yardstick`-like metrics
+
+Unsupervised methods have very different (and more ambiguous) metrics for "success" than supervised methods.  (see above for lots and lots of detail)
```

Comment:
Perhaps we could change from the nomenclature of "performance metrics" to something like "partition characteristics", the former being more of a supervised learning thing.